vcpkg-ports/imagemagick/portfile.cmake

```diff
@@ -30,13 +32,46 @@ vcpkg_configure_make(
     --with-zlib
     --with-zstd
     --with-openexr
-    # FIXME (gdi32 is used in MagickCore which is a C project, while gdi32 is c++)
+    --with-bzlib
+    --with-lcms
+    --with-rsvg
+    # FIXME (gdi+ is used in MagickCore which is a C project, while gdi+ is c++)
     # msvc automatically uses C mode when compiling files with extension "c"
     --without-gdi32
+    --without-autotrace
+    --without-djvu
+    --without-dps
+    --without-flif
+    --without-fpx
+    --without-gslib
+    --without-gvc
+    --without-heic
+    --without-jbig
+    --without-jxl
+    --without-lqr
+    --without-ltdl
+    --without-openjp2
+    --without-perl
+    --without-raqm
+    --without-tiff
+    --without-webp
+    --without-wmf
+    --without-raw # libraw exists but it has a bug with its pkgconfig file
     --disable-docs
 )
 
+# FIXME
```

Comment:
Can you add more detailed explanation here? For FIXME.

Reply:
The short story: The proper fix would be to edit glib port to change the rpath of glib's libraries (glib doesn't want me to live in peace :"D)

The long story: Imagemagick depends on Gio (from Glib). Gio privately depends on gmodule (also from Glib). This means that linking against Gio does not need us to link against gmodule (since gio is already linked against gmodule), but instead gmodule has to be in the runtime search path while linking to resolve all symbols. Hence setting the LD_LIBRARY_PATH environment variable.

This isn't a good solution though, since `vcpkg_install_make` builds both debug and build, and the environment variable is only set for release.

Reply:
Vcpkg had the option `VCPKG_FIXUP_ELF_RPATH` which changes the rpath of installed libraries automatically. I used it in our custom triplet and it fixed the issue for me.